src/XCCDF/xccdf_session.c

```diff
@@ -890,6 +896,25 @@ int xccdf_session_load_oval(struct xccdf_session *session)
 			return 2;
 		}
 
+		if (session->export.thin_results) {
+			struct oval_results_model *res_model = oval_agent_get_results_model(tmp_sess);
+			struct oval_result_directives *dir = oval_directives_model_new();
```

Comment:
Note to self: The pointer types are not compatible! This is undefined behavior and causes crashes...